### ISSUE TITLE
panic when joining one copy of an item and also using the item as values in object

### ISSUE THREAD

**Issue by philpennock** [mikefarah/yq#753]
**Describe the bug**

I might have the query syntax wrong, but this shouldn't yield a panic:

```console
% yq eval '.. | select(has("go-version")).go-version as $g | [{"file": filename, "path": $g|path|join("."), "go-version": $g}]' --  */.github/workflows/* 
panic: runtime error: index out of range [0] with length 0
```

Version of yq: 4.6.1  (v4.6.1-3-gb2186d5)
Operating system: linux (Ubuntu 20.04)
Installed via: `make local install`

The version I had locally failed, so I checked latest git to see if it was fixed yet.

**Input Yaml**

data1.yml:
```yaml
---
foo:
  bar:
    go-version: 1.16
```

**Command**
The command you ran:
```
yq eval '.. | select(has("go-version")).go-version as $g | [{"file": filename, "path": $g|path|join("."), "go-version": $g}]'
```

**Actual behavior**

panic

**Expected behavior**

An error message telling me what I got wrong this time

**Additional context**

I'm using `as $g` because I think using `path | join(".")` did something to reset the value of `.` in the next field of the object.

Ideally there would be an alternative to `path` which emits in a selector format string, such as `.foo[2].bar.baz.go-version`

I'm trying to audit github workflow files and grab both the go-version and the matrix.go list.  Stopping for the night with this repeatable panic.

**Comment by mikefarah** [comment]
Thanks for the clear and detailed issue, had a good play with this, the panic appears to be caused by an operator precedence problem, something to do with using path/join withing `{}` - going to dig deeper and fix that up.

Till then, you can use brackets to avoid the issue:

```
yq eval '.. | select(has("go-version")).go-version | [{"file": filename, "path": (path|join(".")), "go-version": .}]' examples/data1.yaml 
```

**Comment by mikefarah** [comment]
Fixed in 4.6.2
